skill-tree /src/main/java/com/RDS/skilltree/utils/JWTUtils.java

```diff
@@ -0,0 +1,91 @@
+package com.RDS.skilltree.utils;
+
+import io.jsonwebtoken.Claims;
+import io.jsonwebtoken.Jwts;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.security.authentication.AuthenticationCredentialsNotFoundException;
+import org.springframework.stereotype.Component;
+
+import java.security.KeyFactory;
+import java.security.NoSuchAlgorithmException;
+import java.security.interfaces.RSAPrivateKey;
+import java.security.interfaces.RSAPublicKey;
+import java.security.spec.InvalidKeySpecException;
+import java.security.spec.PKCS8EncodedKeySpec;
+import java.security.spec.X509EncodedKeySpec;
+import java.util.Base64;
+
+@Component
+@Slf4j
+public class JWTUtils {
+
+    @Value("${jwt.rds.public.key}")
+    private String publicRDSKeyString;
+    private static KeyFactory keyFactory = null;
+
+    static {
```

Comment:
Rather than making the `KeyFactory` static, you can have `private KeyFactory rsaKeyFactory`.
Then have an init method
```
@PostConstruct
public void init(){
// logic to initialize
}
```

As an alternative, you could do all the work in the constructor.
Additionally, the `RSAPublicKey` for `publicRDSKeyString` will always be the same. So you'd want to perform `convertToRSAPublicKey(publicRDSKeyString)` once during class initialization in order to avoid recomputing the same value every single time.
 
Here's the link to my original comment https://github.com/Real-Dev-Squad/skill-tree-backend/pull/22#discussion_r1391980148